### PR TITLE
Auto-redirect to OAuth when there's only one provider

### PR DESCRIPTION
**Pull request by rnbez**
#### What is the purpose of this pull request?
This PR adds an auto-redirect to an OAuth provider when it is the only login option available for the store. It does it by replacing the `<LoginOptions />` component and informing the user he/she is being redirected to an external site.

>Note: The best use of this functionality is achieved when the store theme is also configured to hide the login form in the Login Page by setting the prop `isInitialScreenOptionOnly` to true.

This PR also adds a loading component to the login options.

#### What problem is this solving?
Some stores want to have a cross-site login using our built-in OAuth solution. In order to improve the user experience, they don't the user to have to open the login modal or page and then click on the OAuth provider option. They would like to be automatic.

#### How should this be manually tested?
To test the app with the redirect, go to [parceiro01's home](https://bezerra--parceiro01.myvtex.com/) or [parceiro01's login page](https://bezerra--parceiro01.myvtex.com/login). This account has only facebook enabled and should redirect you there.
To test the app without the redirect, goto [storecomponents's home](https://bezerra--storecomponents.myvtex.com/) or [storecomponents's login page](https://bezerra--storecomponents.myvtex.com/login). This account has more than one OAuth provider, and *should not* redirect you.

##### Login Modal (Home page) Before 
oauth-redirect-home-before
##### Login Modal (Home page) After
oauth-redirect-home-after

##### Login Page Before 
oauth-redirect-login-2-before
##### Login Page After 
oauth-redirect-login-2-after

#### Types of changes
- [ ] Bug fix (a non-breaking change which fixes an issue)
- [x] New feature (a non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Requires change to documentation, which has been updated accordingly.

### CODE REVIEW

File: messages/context.json [hunk 1]
```diff
@@ -35,6 +35,7 @@
   "store/login.email.placeholder": "store/login.email.placeholder",
```
Comment (brunojdo) [inline messages/context.json]: changelog

File: react/LoginContent.js [hunk 2]
```diff
@@ -182,6 +184,14 @@ class LoginContent extends Component {
     )
   }
 
+  shouldRedirectToOAuth = (loginOptions) => {
```
Comment (brunojdo) [inline react/LoginContent.js]: remove parentheses ⛏

File: react/components/OAuthAutoRedirect.js [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+import { injectIntl, intlShape } from 'react-intl'
+import { Spinner } from 'vtex.styleguide'
+import { AuthService } from 'vtex.react-vtexid'
+
+function OAuthAutoRedirect({ intl, provider, redirect }) {
+  useEffect(() => {
+    redirect()
+  }, [])
+  return (
+    <div className="w-100 flex flex-column pv5">
```
Comment (klzns) [inline react/components/OAuthAutoRedirect.js]: Add a CSS Modules class just like line 15 and 68 of LoginOptions.

File: react/components/Loading.js [hunk 4]
```diff
@@ -0,0 +1,12 @@
+import React from 'react'
+import { Spinner } from 'vtex.styleguide'
+
+function Loading() {
+  return (
+    <div className="w-100 tc c-emphasis pv8">
```
Comment (klzns) [inline react/components/Loading.js]: Same

File: react/components/OAuthAutoRedirect.js [hunk 5]
```diff
@@ -0,0 +1,52 @@
...
+      <p className="tc ph4">
+        {intl.formatMessage(
+          {
+            id: 'store/login.autoRedirect.message',
+          },
+          { provider }
+        )}
+      </p>
+      <div className="self-center c-emphasis">
```
Comment (klzns) [inline react/components/OAuthAutoRedirect.js]: Same

File: react/components/OAuthAutoRedirect.js (hunk 3 above)
Comment (brunojdo) [inline react/components/OAuthAutoRedirect.js]: add cssHandle here

File: react/components/OAuthAutoRedirect.js [hunk 6]
```diff
@@ -0,0 +1,52 @@
...
+    redirect()
+  }, [])
+  return (
+    <div className="w-100 flex flex-column pv5">
+      <p className="tc ph4">
+        {intl.formatMessage(
+          {
+            id: 'store/login.autoRedirect.message',
+          },
+          { provider }
```
Comment (rafarubim) [inline react/components/OAuthAutoRedirect.js]: So it seems the API was not straight forward 😛

File: react/components/OAuthAutoRedirect.js [hunk 7]
```diff
@@ -0,0 +1,52 @@
...
+        <Spinner color="currentColor" size={24} />
+      </div>
+    </div>
+  )
+}
+OAuthAutoRedirect.propTypes = {
+  redirect: PropTypes.func.isRequired,
+  provider: PropTypes.string.isRequired,
+  loading: PropTypes.bool,
+  intl: intlShape,
```
Comment (rafarubim) [inline react/components/OAuthAutoRedirect.js]: .isRequired

File: react/components/OAuthAutoRedirect.js [hunk 8]
```diff
@@ -0,0 +1,52 @@
...
+          loading={loading}
+          provider={provider}
+          redirect={redirectToOAuthPage}
+        />
+      )}
+    </AuthService.OAuthRedirect>
+  )
+}
+Wrapper.propTypes = {
+  provider: PropTypes.string.isRequired,
```
Comment (rafarubim) [inline react/components/OAuthAutoRedirect.js]: intl: intlShape.isRequired,

File: messages/context.json (hunk 1 above)
Comment (rnbez) [reply messages/context.json]: Done

File: react/components/OAuthAutoRedirect.js (hunk 8 above)
Comment (rnbez) [reply react/components/OAuthAutoRedirect.js]: As far as I know, `intlShape` is already the _required_ version.

**Pull request merged**